platform/engineering/design-docs/2020-06-product-slo-tools.md

```diff
@@ -0,0 +1,360 @@
+# Product SLO Tools Design Doc
+
+- **Author(s):** Keifer Furzland
+- **Last Updated:** June 04, 2020
+- **Status:** Draft <!-- | _In Review_ | _Approved_ -->
+- **Approvers:** Dror Matalon \[ \], Andrew Gunsch \[ \], Wyatt Walter \[ \], John Paul Ashenfelter \[ \], Rian Fowler
+    \[ \]
+
+## Overview
+
+> The intended audience for this document is software engineers working on the VSP.
+
+### Objective
+
+The VSP objective is to build pathways for teams at VA to easily launch consistent, high-quality digital products **_informed by data_**. In order for teams to actively work to improve product performance, some sort of performance objective must be defined. This is where service level objectives (SLOs) are valuable.
+
+The objectives of this product include:
+
+- provide processes and tooling for product teams to define and iterate SLOs
+- provide a source of truth for SLO performance
+
+### Background
+
+Engineering effort is a scarce resource, and should be invested in the most important work. A thoughtful, engaged, sophisticated SLO process allows for data-informed decisions about opportunity cost of improved reliability. Another benefit of adopting SLO-based decision making is allowing teams experiment and iterate features more safely, by reducing risk of end-user impact.
+
+Product based SLOs are concerned with the reliability of a specific product on VA.gov.
+
+Example products include:
+
+- [Global Site - Search](https://github.com/department-of-veterans-affairs/va.gov-team/tree/master/products/global/search)
+- [Personalized Experience - Direct Deposit](https://github.com/department-of-veterans-affairs/va.gov-team/tree/master/products/identity-personalization/direct-deposit)
+- [Facilities Locator](https://github.com/department-of-veterans-affairs/va.gov-team/tree/master/products/facilities/facility-locator)
+
+---
+
+**Read more**
+
+- [Product SLO Tools - Product Outline](../../products/platform/product-slo-tools/product-outline.md)
+- [Chapter 2](https://landing.google.com/sre/workbook/chapters/implementing-slos/) of the Google SRE book explains the basics of setting and implementing SLOs
+- Datadog maintains [well written docs](https://docs.datadoghq.com/monitors/service_level_objectives/monitor/) for SLOs
+
+### High Level Design
+
+The high-level design of the Product SLO lifecycle includes these steps:
+
+1. VFS Product team works in conjunction with the VSP Tools teams to define SLOs
```

Comment:
@kfrz I would love to hear more opinions on this, but my take is that we need to set baseline SLO standards that all VA.gov products must live up to, so I'm hesitant to define SLOs w/the VFS team on a per product basis, unless that collaborative definition would be to define service level objectives above and beyond the standards.

Reply:
I agree, we should have reasonable defaults, so long as those defaults are a floor that maintain a reasonable user experience -- but we *should*, in my opinion, make sure we tailor a given product's SLO to the particular user experience, whatever that may be. 

That said we can dictate the initial baseline SLOs and build from there